extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -172,23 +178,35 @@ public void onStop() {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("Stopping JMSSubscriber source");
     }
+
+    jmsLock.unlock();
+
+    try {
+      if (ackMode.equals(AUTO) || ackMode.equals(DUPS_OK) || ackMode.equals(MANUAL)) {
+        session.get().recover();
```

Comment:
> the old transport had the concept of how many consumers to have on each destination. When the source was stopped, the stop logic had to be done on each of those consumers

In this new JMS Listener, a listener means only one JMS consumer, therefore, there is no need to stop multiple consumers.
But also, the stop of the consumers/producers is done when returning the listener's connection `BaseConnectionProvider#onReturn()`

> Shouldn't the ACK logic be done for each of them?

Not sure to understand, but if the Listeners is being stopped, we can not guarantee that the in-flight messages have been correctly consumed, so we can't execute an Acknowledgement over the session.